c10/CMakeLists.txt

```diff
@@ -0,0 +1,49 @@
+project(c10 LANGUAGES CXX)
+
+set(LIB_SOURCES
+        dummy.cpp
+)
+
+set(TEST_SOURCES
+        dummy_test.cpp
+)
+
+set(COMMON_WARNING_OPS
+        -Wall
+        -Wextra
+        -Wold-style-cast
+        -Wno-missing-braces
+        -Wcast-align
+        -Wcast-qual
+        -Wctor-dtor-privacy
+        -Wdisabled-optimization
+        -Wformat=2
+        -Winit-self
+        -Wmissing-include-dirs
+        -Woverloaded-virtual
+        -Wredundant-decls
+        -Wshadow
+        -Wsign-promo
+        -Wstrict-overflow=5
+        -fdiagnostics-show-option
+        -Wconversion
+        -Wpedantic
+        -Wno-gnu-zero-variadic-macro-arguments
+        -Werror
+)
+
+set(LIB_WARNING_OPTS
+        -Wundef  # lib-only because it fails in gtest headers
+        ${COMMON_WARNING_OPS}
+)
+
+set(TEST_WARNING_OPS
+        ${COMMON_WARNING_OPS}
+)
+
+add_library(${PROJECT_NAME} OBJECT ${LIB_SOURCES})
+target_compile_options(${PROJECT_NAME} PRIVATE ${LIB_WARNING_OPTS})
+
+add_executable(${PROJECT_NAME}_test ${TEST_SOURCES} $<TARGET_OBJECTS:${PROJECT_NAME}>)
+target_compile_options(${PROJECT_NAME}_test PRIVATE ${TEST_WARNING_OPTS})
```

Comment:
and since you don't have the typo here, I'm skeptical that this has been tested :)

can you integrate it into the caffe2 build so that it's not completely dead/untested code?